resharper/resharper-unity/src/Yaml/Psi/DeferredCaches/UnityAssetsCacheUpdater.cs

```diff
@@ -26,14 +28,16 @@ public UnityCacheUpdater(Lifetime lifetime, ISolution solution, DeferredCacheCon
                     if (handler.HasNew && handler.HasOld && handler.New == handler.Old)
                         return;
 
-                    locks.ExecuteOrQueueReadLockEx(lifetime, "YamlParsingStateChange", () =>
+                    locks.ExecuteOrQueueReadLockEx(lifetime, "Unity asset indexing cache state change", () =>
                     {
                         var psiSourceFiles = module.SourceFiles.ToList();
                         if (psiSourceFiles.Any())
                         {
                             locks.ExecuteWithWriteLock(() =>
                             {
                                 deferredCacheController.Invalidate<UnityAssetsCache>();
+
+                                // TODO: Why is this being run through ChangeManager?
                                 changeManager.ExecuteAfterChange(() =>
```

Comment:
Which way will be better for updating SimpleICaches for asset source files?

Reply:
The update bit is fine, I don't get why it's done inside `ExecuteAfterChange`. I don't see how this could be called while executing another change.